### ISSUE TITLE
FALSE POSITIVE: NP_NULL_ON_SOME_PATH ignores a known-true instanceof check

### ISSUE THREAD

**Issue by lynae99** [spotbugs/spotbugs#4273]
Hi, I found a false positive in SpotBugs 4.10.3 when the runtime object is exactly Derived, so result is assigned.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs NP_NULL_ON_SOME_PATH`

**Minimal reproducer**

```java
class SpotBugsKnownInstanceof {
  class Base {}
  final class Derived extends Base {}
  void test() {
    Base value = new Derived();
    Object result = null;
    if (value instanceof Derived) result = new Object();
    result.toString();
  }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-np-null-on-some-path-fp-known-instanceof.java
spotbugs -textui -effort:max -low -visitors FindNullDeref classes
```

**Current behavior**

SpotBugs reports `NP_NULL_ON_SOME_PATH` at line 8.

**Expected behavior**

SpotBugs should not report `NP_NULL_ON_SOME_PATH` because the runtime object is exactly Derived, so result is assigned.
